WISE-Community/WISE
# Moved applicationContexts config from xml -> java

How to test:

1. Update application.properties. See changes made to application_sample.properties
   1. add ```spring.servlet.multipart.max-file-size=100MB```
   2. add ```spring.jpa.hibernate.use-new-id-generator-mappings=false```
   3. set the new spring.mail properties
```
spring.mail.host=smtp.gmail.com
spring.mail.port=587
spring.mail.username=
spring.mail.password=
spring.mail.properties.mail.smtp.auth=true
spring.mail.properties.mail.smtp.starttls.enable=true
```
2. Run mvn test and make sure that all tests pass
3. Test WISE4 and WISE5 and make sure that they behave as before. Test these in particular:
   1. Log in with google and non-google student/teacher
   2. Share projects/runs
   3. Upload files as student, author, admin
   4. Log in as another user 
   5. Send contact email 

Closes #1982
